tests/storage/mpathconf_test.py

```diff
@@ -172,3 +201,46 @@ def test_mpath_empty_conf(fake_mpath_conf):
         revision=mpathconf.REVISION_MISSING,
         private=False,
     )
+
+
+@pytest.fixture
+def fake_mpathd_config(request, monkeypatch):
+    def _factory(mpath_output):
+        def mockreturn(*args):
+            return mpath_output
+
+        monkeypatch.setattr(
+            mpathconf.multipath, 'show_config_local', mockreturn)
+
+    return _factory
+
+
+@pytest.mark.parametrize("test_input,section", [
+    (["yes", "no", "no"], "defaults"),
+    (['"yes"', '"no"', '"no"'], "defaults"),
+    (["no", "yes", "no"], "overrides"),
+    (['"no"', '"yes"', '"no"'], "overrides"),
+    (["no", "no", "yes"], "multipath"),
+    (['"no"', '"no"', '"yes"'], "multipath"),
+])
+def test_conf_ufn_enabled(fake_mpathd_config, test_input, section):
+    defaults_value, overrides_value, mpath_value = test_input
+    fake_mpathd_config(FAKE_MPATHD_OUTPUT.substitute(
+        {"defaults_value": defaults_value,
+         "overrides_value": overrides_value,
+         "mpath_value": mpath_value}))
+    with pytest.raises(exception.InvalidConfiguration) as e:
+        mpathconf.check_local_config()
+    assert section == e.value.context["section"].name
+    invalid_pair = str(mpathconf.ConfPair('user_friendly_names', 'yes'))
+    assert invalid_pair in str(e.value)
```

Comment:
I think we need simple test of the parser before we test the code validating the parsed tree.